memgraph-graphRAG/graphRAG.ipynb

```diff
@@ -0,0 +1,797 @@
+    "    prompt = RAG_prompt(question, relevant_data)\n",
+    "    response = asyncio.run(get_response(client, prompt))\n",
+    "    print(\"The response is:\")\n",
+    "    print(response)\n",
+    "\n",
+    "    driver.close()\n",
+    "\n",
+    "\n",
+    "if __name__ == \"__main__\":\n",
+    "    main()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "Here are a few examples of questions and answers: \n",
+    "\n",
+    "**To whom was Viserys Targaryen loyal to?**\n",
+    "\n",
+    "The response is:\n",
+    "Based on the provided data, Viserys Targaryen was loyal to House Targaryen. This information is derived from the relationships indicating that Viserys Targaryen was loyal to House Targaryen and the connections between them\n",
+    "\n",
+    "**Who killed Viserys Targaryen in Game of thrones?**\n",
+    "\n",
+    "The response is:\n",
+    "\n",
+    "**Based on the provided relevance expansion data, Khal Drogo killed Viserys Targaryen in \"Game of Thrones.\" This information is inferred from the relationship where Khal Drogo is linked to Viserys Targaryen with the action of being \"KILLED\" by Khal Drogo. The data does not show any other character directly killing Viserys Targaryen.\n",
+    "**\n",
+    "\n",
+    "**\"What was the weapon used to kill Viserys Targaryen in Game of Thrones?\"**\n",
+    "\n",
+    "The response is: \n",
+    "\n",
+    "Based on the provided data, the weapon used to kill Viserys Targaryen in Game of Thrones was not explicitly mentioned. The data only shows that Khal Drogo was involved in the killing of Viserys Targaryen. There is no specific mention of the weapon used in the relevance expansion data. Therefore, I do not have enough information to answer the question about the weapon used to kill Viserys Targaryen.\n",
+    "\n",
+    "This response is wrong, there is a method mentioned, not weapon, but LLM didn't catch the context due to different naming. \n",
+    "\n",
+    "**\"Who betrayed Viserys Targaryen in Game of Thrones?\"**\n",
+    "\n",
+    "Based on the provided data, Khal Drogo betrayed Viserys Targaryen in Game of Thrones by killing him. This conclusion is drawn from the relationship between Khal Drogo and Viserys Targaryen where it is stated that Khal Drogo killed Viserys Targaryen.\n",
+    "\n",
+    "This response is based on the killing relationship, but betrayal could have different consequences. \n",
+    "\n",
+    "Let's expand this knowledge. \n",
+    "\n",
+    "**\"Who betrayed Viserys Targaryen in Game of Thrones?\"**\n",
+    "\n",
+    "Based on the provided data, Khal Drogo betrayed Viserys Targaryen in Game of Thrones. This conclusion is derived from the relationship between Viserys Targaryen and Khal Drogo, where Khal Drogo is connected to Viserys Targaryen through the 'BETRAYED_BY' relationship, indicating that Khal Drogo betrayed Viserys Targaryen."
```

Comment:
There is a disconnect between this part and the last part in Ante's section. It is unclear what the wrong answer was and how this helped.